src/utils/tree_node.C

```diff
@@ -457,25 +460,30 @@ unsigned int TreeNode<N>::n_active_bins() const
 
 
 template <unsigned int N>
-const Elem* TreeNode<N>::find_element(const Point& p, const std::set<subdomain_id_type> *allowed_subdomains) const
+const Elem*
+TreeNode<N>::find_element
+(const Point& p,
+ const std::set<subdomain_id_type> *allowed_subdomains,
+ Real relative_tol) const
 {
   if (this->active())
     {
       // Only check our children if the point is in our bounding box
       // or if the node contains infinite elements
-      if (this->bounds_point(p) || this->contains_ifems)
+      if (this->bounds_point(p, relative_tol) || this->contains_ifems)
```

Comment:
This does make the backup find_element_in_children search into a null operation...

But I'm not yet convinced, once we get the other issues straightened out, that that fallback should exist.  If we've searched all child nodes that contain the point, but we didn't find the element that contains the point, then that seems to be a failure in the tree construction, and I'd like to first try to fix the construction rather than paper over the failure with more-expensive fallback searches.

On the other hand, fixing the flaw in general would require being able to find a true bounding box for higher-order geometric elements.  If we can't figure out how to do that, then the expensive fallback here would still be preferable to the vastly-more-expensive linear search fallback, and the right thing to do would be to delete that bounding box check.